src/basics.md

```diff
@@ -470,6 +471,48 @@ fn main() {
 }
 ```
 
+[ex-phone]: #ex-phone
+<a name="ex-phone"></a>
+## Extract phone numbers from text
+
+Attempts to process a user input as a US phone number.  Essentially, the
+expression is checking that there are 7 digits that come in groups of 3 3 4,
+removing any extra characters.  The phone number is then formatted for display.
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate regex;
+
+use std::io::stdin;
+use regex::Regex;
+#
+# error_chain!{
+#     foreign_links {
+#         Regex(regex::Error);
+#         Io(std::io::Error);
+#     }
+# }
+
+fn run() -> Result<()> {
+    println!("Input a phone number");
+    let mut phone_input = String::new();
+    // Regular expression is looking for 3 groups of digits, capturing them
+    let re = Regex::new(r".*([2-9]\d{2}).*(\d{3}).*(\d{4})")?;
+    stdin().read_line(&mut phone_input)?;
+    // Make sure the regular expression captured something
+    if let Some(captures) = re.captures(&phone_input){
+        let formatted_phone = format!("({}) {}-{}", &captures[1], &captures[2], &captures[3]);
+        println!("Formatted like a boss: {}", formatted_phone);
```

Comment:
how about just printing these out instead of formatting? also no need to add additional noise to the output. lets keep the example short and sweet so it can be copy pasted.

Also I leave it up to you if you would consider using PhoneNumber struct with display (please see the external process example)